### PR TITLE
LPS-189055 Add a mechanism for massive regeneration of Video and Audio previews

### PR DESCRIPTION
**Pull request by robertoDiaz**

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:reevaluate:1397401_6063

File: modules/apps/document-library/document-library-preview-api/src/main/java/com/liferay/document/library/preview/background/task/BasePreviewBackgroundTaskExecutor.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+					}
+				}
+			});
+
+		try {
+			actionableDynamicQuery.performActions();
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-preview-api/src/main/java/com/liferay/document/library/preview/background/task/BasePreviewBackgroundTaskExecutor.java:122]: Can you make this method throw the PortalException and log it in the `execute` method?

File: modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+)
+public class AudioPreviewBackgroundTaskExecutor
+	extends BasePreviewBackgroundTaskExecutor {
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		dlFileEntryConfiguration = ConfigurableUtil.createConfigurable(
+			DLFileEntryConfiguration.class, properties);
+	}
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java:48]: Why not move this to the base class? It is only used there (this applies to both derived classes).

File: modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	@Override
+	protected void generatePreview(FileVersion fileVersion) throws Exception {
+		audioProcessor.generateAudio(null, fileVersion);
+	}
+
+	@Override
+	protected String[] getMimeTypes() {
+		Set<String> audioMimeTypes = audioProcessor.getAudioMimeTypes();
+
+		return audioMimeTypes.toArray(new String[0]);
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java:59]: Why not use `ArrayUtil.toStringArray`?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java (hunk 2 above)
Comment (robertoDiaz) [reply modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java]: Done

File: modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java (hunk 3 above)
Comment (robertoDiaz) [reply modules/apps/document-library/document-library-preview-audio/src/main/java/com/liferay/document/library/preview/audio/internal/background/task/AudioPreviewBackgroundTaskExecutor.java]: done

File: modules/apps/document-library/document-library-preview-api/src/main/java/com/liferay/document/library/preview/background/task/BasePreviewBackgroundTaskExecutor.java (hunk 1 above)
Comment (robertoDiaz) [reply modules/apps/document-library/document-library-preview-api/src/main/java/com/liferay/document/library/preview/background/task/BasePreviewBackgroundTaskExecutor.java]: Done

**Review by adolfopa — APPROVED** [review]

**Comment by adolfopa** [comment]
ci:forward

**Comment by austinchiang** [comment]
Test failures unrelated

**Comment by austinchiang** [comment]
ci:forward:force

**Pull request closed without merging**